### PR TITLE
#83820 Tab management (part 1)

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: extending-valtimo/tabs/customizing-case-list-tabs.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# Customizing case list tabs
+# ⚠️**WARNING**⚠️
```
Comment (ivo-ritense) [inline extending-valtimo/tabs/customizing-case-list-tabs.md:1]: This page is for the case list. This part only is not being replaced.

File: extending-valtimo/tabs/customizing-case-tabs.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-# Customizing case details tabs
+# ⚠️**WARNING**⚠️
```
Comment (marijnritense) [inline extending-valtimo/tabs/customizing-case-tabs.md:1]: I'll just post this here, but it goes for every warning: Please use the gitbook recommended way of doing warning hints. See: https://docs.gitbook.com/content-creation/blocks/hint

File: reference/modules/case.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Case
+
+## Case tabs types
+
+The following case types are supported by Valtimo.
+
+| Type         | Type key | Description                                                                                                              |
+|--------------|----------|:-------------------------------------------------------------------------------------------------------------------------|
+| **Standard** | standard | A collection of tabs is available that support different Valtimo features. The tabs that are available are listed below. |
+| **Form.io**  | formio   | Form.io tabs can be configured to show a form.io form.                                                                   |
```
Comment (mritense) [inline reference/modules/case.md:10]: Form.io form

File: reference/modules/case.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| **Custom**   | custom   | Custom tabs can be configured to show a custom Angular component.                                                        |
+
+
+### Standard tabs
+
+| Type                | Type key        | Description                                                                                                                                                             |
+|---------------------|-----------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Summary**         | summary         | Shows a single form.io form with the name: '<case-key>.summary'.                                                                                                        |
```
Comment (mritense) [inline reference/modules/case.md:18]: Form.io

File: using-valtimo/case/case-tabs.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+### Standard tabs
+
+Valtimo comes with a number of tabs that support different features. These range from showing a summary form, or showing
+an audit log to showing the current state of the BPMN process.
+
+The standard tabs that are available are listed on the [reference page](/reference/modules/case.md#standard-tabs).
+
+### Form.io tabs
+
+Form.io tabs are tabs that show a single form.io form as the tab content. Any form.io form that has been registered
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:27]: Form.io form

File: using-valtimo/case/case-tabs.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+with valtimo can be chosen. This is purely meant to display information, as there is no way to submit the information
+even if a submit button has been added to the form definition.
+
+### Custom tabs
+
+When you want tab with functionality that is not covered by the other tab types you can create a custom angular
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:33]: Angular

File: using-valtimo/case/case-tabs.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+component that is used as a case tab. These angular components have to be registered in the
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:34]: Angular

File: using-valtimo/case/case-tabs.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+angular application as a potential tab.
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:35]: Angular

File: using-valtimo/case/case-tabs.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+#### **`app.module.ts`**
+
+```typescript
+...
+// import CASE_TAB_TOKEN
+import { CASE_TAB_TOKEN } from '@valtimo/dossier';
+...
+// import your Custom Component from wherever you have defined it
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:44]: import your custom component

File: using-valtimo/case/case-tabs.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+4. Click on the 'Add tab' button
+
+   ![Add tab button](img/add-tab-button.png)
+
+5. Select which tab type you want to configure
+
+   ![Example of selecting a tab type](img/tab-select-modal.png)
+
+6. Fill in the form. If no name is set, then the key will be used for translation (in your translation files, add the key under 'dossier.tabs')
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:80]: `dossier.tabs.{YOUR_TAB_KEY}`

File: using-valtimo/case/case-tabs.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+![Example of disabled tab type](img/tab-select-disabled.png)
+
+This can occur for one of two reasons:
+
+- You have already configured all possible tabs of a type
+- There are no tabs provided for configuration (i.e. no form definitions or no configurations for custom tabs)
+
+## Auto deploying case tabs
+
+Case tabs can be auto-deployed from a json file at startup. This is useful to keep the case tab configuration identical
```
Comment (mritense) [inline using-valtimo/case/case-tabs.md:97]: JSON

**Review by mritense — APPROVED** [review]

**Pull request merged**